File: data-structure/recursion/countdown.c
// 카운트다운하는 함수
#include <stdio.h>

void countdown(int i)
{
	printf("%d\n", i);

	// 기본 단계 (base case)
	if (i <= 0)
		return;
	// 재귀 단계 (recursive case)
	else
		countdown(i - 1);
}

int main(void)
{

	countdown(5);

	return 0;
}

File: data-structure/recursion/countdown.py
# 카운트다운하는 함수

def countdown(i):
  # 기본 단계 (base case)
  if i <= 0:
    return 0
  # 재귀 단계 (recursive case)
  else:
    print(i)
    return countdown(i-1)

countdown(5)
